src/Components/NewPost.js

```diff
@@ -14,17 +14,19 @@ import { useSelector, useDispatch } from 'react-redux'
 
 import { randomImgAPI } from '../utils/api'
 
-function NewPost(props) {
-	const { modal, setModal } = props
+import { useModal } from '../Providers/ModalProvider'
 
+function NewPost() {
 	//add img source later in backed db
 	const user = useSelector(state => state.auth.user)
 
-	const { dp, username } = user
+	const { modal, setModal } = useModal()
 
-	const imgSrc = !!dp ? dp : randomImgAPI
+	const { dp, username } = user
 
-	console.log(imgSrc)
+	const imgSrc = !!dp
```

Comment:
query: why are you using `!!dp` here?

Reply:
it is that I have send aliased profilePic as dp in backend and I send that on initial load. So if the user hasnt yet updated his profilePic (i wil work on this later if i get time)after registering a random pic is shown